website/docs/advanced/override_grammar/extended.md

```diff
@@ -0,0 +1,237 @@
+ 
+```python title="Signature"
+def interval(start: Union[int, float], end: Union[int, float]) -> IntervalSweep:
+    """
+    A continuous interval between two floating point values.
+    value=interval(x,y) is interpreted as x <= value < y
+    """
+```
+```yaml title="Examples"
+#interval(1.0,5.0)  # 1.0 <= x < 5.0
+#interval(1,5)      # 1.0 <= x < 5.0, auto-cast to floats
+```
+
+### Tag
+Tagging allows advanced sweepers to add arbitrary metadata to a sweep.
+```python title="Signature"
+def tag(*args: Union[str, Union[Sweep]], sweep: Optional[Sweep] = None) -> Sweep:
+    """
+    Tags the sweep with a list of string tags.
+    """
+```
+```yaml title="Examples"
+#tag(log,interval(0,1))          # 1.0 <= x < 1.0, tags=[log]
+#tag(foo,bar,interval(0,1))      # 1.0 <= x < 1.0, tags=[foo,bar]
+```
+
+## Reordering lists and sweeps
+
+### sort
+```python title="Signature"
+def sort(
+    *args: Union[ElementType, ChoiceSweep, RangeSweep],
+    sweep: Optional[Union[ChoiceSweep, RangeSweep]] = None,
+    list: Optional[List[Any]] = None,
+    reverse: bool = False,
+) -> Any:
+    """
+    Sort an input list or sweep.
+    reverse=True reverses the order
+    """
+```
+```yaml title="Examples"
+# sweep
+sort(1,3,2)                         # ChoiceSweep(1,2,3)
+sort(1,3,2,reverse=true)            # ChoiceSweep(3,2,1)
+sort(choice(1,2,3))                 # ChoiceSweep(1,2,3)
+sort(sweep=choice(1,2,3))           # ChoiceSweep(1,2,3)
+sort(choice(1,2,3),reverse=true)    # ChoiceSweep(3,2,1)
+sort(range(10,1))                   # range in ascending order
+sort(range(1,10),reverse=true)      # range in descending order
+
+# lists
+sort([1,3,2])                       # [1,2,3]
+sort(list=[1,3,2])                  # [1,2,3]
+sort(list=[1,3,2], reverse=true)    # [3,2,1]
+
+# single value returned as is
+sort(1)                             # 1
+```
+
+### shuffle
+```python title="Signature"
+def shuffle(
+    *args: Union[ElementType, ChoiceSweep, RangeSweep],
+    sweep: Optional[Union[ChoiceSweep, RangeSweep]] = None,
+    list: Optional[List[Any]] = None,
+) -> Union[List[Any], ChoiceSweep, RangeSweep]:
+    """
+    Shuffle input list or sweep (does not support interval)
+    """
+```
+```yaml title="Examples"
+shuffle(a,b,c)                                       # shuffled a,b,c
+shuffle(choice(a,b,c)), shuffle(sweep=choice(a,b,c)) # shuffled choice(a,b,c)
+shuffle(range(1,10))                                 # shuffled range(1,10)
+shuffle([a,b,c]), shuffle(list=[a,b,c])              # shuffled list [a,b,c] 
+```
+
+## Type casting
+You can cast values, and sweeps to int, float, bool or str.
+```yaml title="Example"
+int(3.14)                  # 3 (int)
+int(value=3.14)            # 3 (int)
+float(10)                  # 10.0 (float)
+str(10)                    # "10" (str)
+bool(1)                    # true (bool)
+float(range(1,10))         # range(1.0,10.0)
+str([1,2,3])               # ['1','2','3']
+str({a:10})                # {a:'10'}
```

Comment:
str on a dict casts the value but not the key?

Reply:
correct.